operations/helm/charts/alloy/templates/cluster_service.yaml

```
@@ -29,5 +29,9 @@ spec:
       port: {{ $portMap.port }}
       targetPort: {{ $portMap.targetPort }}
       protocol: {{ coalesce $portMap.protocol "TCP" }}
+      {{- if not (empty $portMap.appProtocol) }}
```

Could this be simplified to:
```suggestion
      {{- if $portMap.appProtocol }}
```

---

Would make it easier to read. But isn't  {{- if not (empty $portMap.appProtocol) }} better if you want to ensure that $portMap.appProtocol is not only present but also non-empty? 

If $portMap.appProtocol is undefined or empty, the first version (empty) will prevent any unexpected issues by skipping the block, while the second version ($portMap.appProtocol) might run into problems if the variable is not properly initialized.

Chances are slim, so I am (in the end) happy with both ❤️